qiskit/circuit/quantumcircuit.py

```diff
@@ -4553,17 +4553,20 @@ def barrier(self, *qargs: QubitSpecifier, label=None) -> InstructionSet:
 
     def delay(
         self,
-        duration: ParameterValueType,
+        duration: Union[ParameterValueType, expr.Expr],
         qarg: QubitSpecifier | None = None,
-        unit: str = "dt",
+        unit: str | None = None,
     ) -> InstructionSet:
         """Apply :class:`~.circuit.Delay`. If qarg is ``None``, applies to all qubits.
         When applying to multiple qubits, delays with the same duration will be created.
 
         Args:
-            duration (int or float or ParameterExpression): duration of the delay.
+            duration (Object):
```

Comment:
FWIW you can just drop `(Object)` here it's all superseded by the type hinting in the docs.